api/src/main/java/org/xwiki/apps/googleapps/internal/CookieAuthenticationPersistenceImpl.java

```diff
@@ -89,38 +77,40 @@
     private static final String EQUAL_SIGN = "=";
     private static final String UNDERSCORE = "_";
 
-    /* Default Cookie TTL : 3 days */
-    private static final COOKIE_TTL = 259200
-
-    private Log logger = LogFactory.getLog(CookieAuthenticationPersistenceStoreTools.class);
+    @Inject
+    private Logger logger;
 
     private XWikiContext context;
-    private XWikiRequest request;
+
+    @Inject
+    private ComponentManager componentManager;
 
     private String cookiePfx;
     private String cookiePath;
     private String[] cookieDomains;
-    private int cookieMaxAge;
+    private long cookieMaxAge;
     private Cipher encryptionCipher;
     private Cipher decryptionCipher;
 
 
     /**
-     * Initialize the tool
-     * @context XWiki Context
-     * @cookieMaxAge Time To Live of the created cookies in scd
+     * Initialize the tool.
+     * @param context XWiki Context
+     * @param cookieMaxAge Time To Live of the created cookies in scd
+     * @throws XWikiException in case of trouble
+     * @since 3.0
      */
-    public void initialize(XWikiContext context, int cookieMaxAge) throws InitializationException
+    @Unstable
+    public void initialize(XWikiContext context, long cookieMaxAge) throws XWikiException
     {
         this.context = context;
```

Comment:
You should use ``Provider<XWikiContext>`` to access the context. Storing a reference to the XWiki context is not a good idea, even if this component is instantiated per lookup, because it can become obsolete easily (on each request).

Reply:
Done with the exception that I cast the component-provided GoogleAppsManager to the GoogleAppsManagerImpl in this package: This allows package-internal methods to be used and not pollute the public API.